src/nvim/window.c

```diff
@@ -676,9 +676,17 @@ void win_config_float(win_T *wp, FloatConfig fconfig)
 
   wp->w_float_config = fconfig;
 
+  bool has_border = wp->w_floating && wp->w_float_config.border;
+  for (int i = 0; i < 4; i++) {
+    wp->w_border_adj[i] =
+      has_border && wp->w_float_config.border_chars[2 * i+1][0];
+  }
+
   if (!ui_has(kUIMultigrid)) {
-    wp->w_height = MIN(wp->w_height, Rows-1);
-    wp->w_width = MIN(wp->w_width, Columns);
+    wp->w_height = MIN(wp->w_height, 
+		       Rows - (1 + wp->w_border_adj[0] + wp->w_border_adj[2]));
```

Comment:
I would have `Rows - 1 - (wp->w_border_adj[0] + wp->w_border_adj[2])` as the -1 is unrelated.

Reply:
Fixed